test/core/expectations/compound_data/test_expectations_cursor.pkb

```diff
@@ -1041,5 +1041,516 @@ Rows: [ 2 differences ]%
+%Join key /*/OWNER does not exists in actual%]';
+    l_actual_message := ut3.ut_expectation_processor.get_failed_expectations()(1).message;
+    --Assert
+    ut.expect(l_actual_message).to_be_like(l_expected_message);
+  end;
+  
+  procedure cur_joinby_comp_twocols_nokey is
+    l_actual   SYS_REFCURSOR;
+    l_expected SYS_REFCURSOR;
+    l_actual_message   varchar2(32767);
+    l_expected_message varchar2(32767);
+  begin
+    --Arrange
+    open l_actual   for select rownum as rn, 'x' SOME_COL  from dual a connect by level < 4;
+    open l_expected for select rownum as rn, 'x' SOME_COL  from dual a connect by level < 4;
+     
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by(ut3.ut_varchar2_list('OWNER,USER_ID'));
+    --Assert
+ l_expected_message := q'[%Actual: refcursor [ count = 3 ] was expected to equal: refcursor [ count = 3 ]%
+Diff:%
+%Unable to join sets:%
+%Join key /*/OWNER does not exists in expected%
+%Join key /*/USER_ID does not exists in expected%
+%Join key /*/OWNER does not exists in actual%
+%Join key /*/USER_ID does not exists in actual%]';
+    l_actual_message := ut3.ut_expectation_processor.get_failed_expectations()(1).message;
+    --Assert
+    ut.expect(l_actual_message).to_be_like(l_expected_message);
+  end;
+  
+  procedure cursor_joinby_compare_exkey is
+    l_actual   SYS_REFCURSOR;
+    l_expected SYS_REFCURSOR;
+    l_actual_message   varchar2(32767);
+    l_expected_message varchar2(32767);
+  begin
+    --Arrange
+    open l_actual   for select rownum as rn, 'x' SOME_COL  from dual a connect by level < 4;
+    open l_expected for select rownum as rn, 'x' SOME_COL  from dual a connect by level < 4;
+     
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by('SOME_COL').exclude('SOME_COL');
+    --Assert
+ l_expected_message := q'[%Actual: refcursor [ count = 3 ] was expected to equal: refcursor [ count = 3 ]%
+Diff:%
+%Unable to join sets:%
+%Join key /*/SOME_COL does not exists in expected%
+%Join key /*/SOME_COL does not exists in actual%]';
+    l_actual_message := ut3.ut_expectation_processor.get_failed_expectations()(1).message;
+    --Assert
+    ut.expect(l_actual_message).to_be_like(l_expected_message);
+  end;
+  
+  procedure cur_joinby_comp_twocols_exkey is
+    l_actual   SYS_REFCURSOR;
+    l_expected SYS_REFCURSOR;
+    l_actual_message   varchar2(32767);
+    l_expected_message varchar2(32767);
+  begin
+    --Arrange
+    open l_actual   for select rownum as rn, 'x' SOME_COL  from dual a connect by level < 4;
+    open l_expected for select rownum as rn, 'x' SOME_COL  from dual a connect by level < 4;
+     
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by(ut3.ut_varchar2_list('RN,SOME_COL')).exclude('RN');
+    --Assert
+ l_expected_message := q'[%Actual: refcursor [ count = 3 ] was expected to equal: refcursor [ count = 3 ]%
+Diff:%
+%Unable to join sets:%
+%Join key /*/RN does not exists in expected%
```

Comment:
Can we remove the leading `/*/` from the message?
That way we get actual column name rather than XPATH.
Not sure if it makes sense but probably 98% of use-cases will be joins by regular columns (not nested elements)

Reply:
Yeah we can make that assumption just have to updated docs to mention that.